nf_core/lint.py

```diff
@@ -335,14 +366,58 @@ def check_version_consistency(self):
         self.passed.append((7, "Version tags are numeric and consistent between container, release tag and config."))
 
 
+    def check_conda_env_yaml(self):
+        """ Check the conda environment file - that versions are pinned and are the latest version """
+
+        if 'environment.yml' not in self.files:
+            return
+
+        # Check that the environment name matches the pipeline name
+        if self.conda_config['name'] != 'nfcore-{}'.format(self.pipeline_name):
+            self.failed.append((8, "Conda environment name is not consistent ('{}', should be 'nfcore-{}')".format(self.conda_config['name'], self.pipeline_name)))
+        else:
+            self.passed.append((8, "Conda environment name was consistent ('{}')".format(self.conda_config['name'])))
+
+        # Check conda dependency list
+        for dep in self.conda_config.get('dependencies', []):
+            if isinstance(dep, str):
+                # Check that each dependency has a verion number
+                try:
+                    assert dep.count('=') == 1
+                except:
+                    self.failed.append((8, "Conda dependency did not have pinned version number: {}".format(dep)))
+                else:
+                    self.passed.append((8, "Conda dependency had pinned version number: {}".format(dep)))
+
+                    # Check if each dependency is the latest available version
+                    depname, depver = dep.split('=', 1)
+                    dep_channels = self.conda_config.get('channels', [])
+                    if '::' in depname:
+                        dep_channels = [depname.split('::')[0]]
+                        depname = depname.split('::')[1]
+                    for ch in reversed(dep_channels):
+                        response = requests.get('https://api.anaconda.org/package/{}/{}'.format(ch, depname))
+                        if response.status_code == 200:
+                            dep_json = response.json()
+                            last_ver = dep_json.get('latest_version')
+                            if last_ver is not None and last_ver != depver:
+                                self.warned.append((8, "Conda package is not latest available: {}, {} available".format(dep, last_ver)))
+                            else:
+                                self.passed.append((8, "Conda package is latest available: {}".format(dep)))
+                            break
+                    else:
+                        self.failed.append((8, "Could not find Conda dependency using the Anaconda API: {}".format(dep)))
+
```

Comment:
Did you check what happens if you get a timeout?

Reply:
No I didn't - I guess I could add a try / except for that.. (could copy example from [MultiQC](https://github.com/ewels/MultiQC/blob/062041051854f41c8e0387ed14149afe98c71ae0/multiqc/utils/megaqc.py#L87-L94)). But how will we add tests for that scenario?! 😱 

Maybe this is a rare enough occurence that we can just accept a nasty crash if it happens?

Reply:
Exactly, except this error: `equests.exceptions.Timeout`

> But how will we add tests for that scenario?! 

The issue is that this part of the code is coupled tightly to the surrounding source code. You could instead define the timout explicetly in `nf_core.py` and the create the lint object with the timout as argument. Or you do that for the function as an additional argument, and for the test call this function with a timout value very small (i.e. 0.001), too quick for the server to respond -> timout :)

Reply:
like `check_conda_env_yaml(self, timeout=10):` -> test then with `check_conda_env_yaml(self, timeout=0.01):`

Reply:
Implemented 👍

Reply:
:sunglasses: